ESP32: Failing to build project

Hi! 

I'm trying to build and flash the hello world on a ESP32, following the docs[0], and it fails because it cannot find the `xtensa/tie/xt_core.h` file. Digging a bit, theres no `arch/xtensa/tie` dir at all.

It looks like that dependency was added here  https://github.com/zephyrproject-rtos/zephyr/commit/a2248782a25e205be9b37a03fc47131856ba406d to `arch/xtensa/core/cpu_idle.c`

I created a Dockerfile[1] to reproduce the problem. Am I missing something?

Thanks!

0_ http://docs.zephyrproject.org/boards/xtensa/esp32/doc/esp32.html
1_ https://paste.debian.net/1040302/

Build output:
```
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.6", minimum required is "3.4") 
-- Selected BOARD esp32
Zephyr version: 1.13.0
Parsing Kconfig tree in /root/zephyr/Kconfig
Using /root/zephyr/boards/xtensa/esp32/esp32_defconfig as base
Merging /root/zephyr/samples/hello_world/prj.conf
warning: ENTROPY_ESP32_RNG (defined at drivers/entropy/Kconfig.esp32:7) was assigned the value 'y'
but got the value 'n'. You can check symbol information (including dependencies) in the 'menuconfig'
interface (see the Application Development Primer section of the manual), or in the Kconfig
reference at http://docs.zephyrproject.org/reference/kconfig/CONFIG_ENTROPY_ESP32_RNG.html (which is
updated regularly from the master branch). See the 'Setting configuration values' section of the
Board Porting Guide as well.
-- Generating zephyr/include/generated/generated_dts_board.h
-- Cache files will be written to: /root/.cache/zephyr
CMake Warning at /root/zephyr/boards/common/esp32.board.cmake:5 (message):
  Setting ESP_IDF_PATH in the environment is deprecated.  Use cmake
  -DESP_IDF_PATH=...  instead.
Call Stack (most recent call first):
  /root/zephyr/boards/xtensa/esp32/board.cmake:1 (include)
  /root/zephyr/cmake/app/boilerplate.cmake:299 (include)
  CMakeLists.txt:3 (include)


-- The C compiler identification is GNU 5.2.0
-- The CXX compiler identification is GNU 5.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /root/esp/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
-- Performing Test toolchain_is_ok
-- Performing Test toolchain_is_ok - Success
-- 
*******************************
*** NOTE TO KCONFIG AUTHORS ***
*******************************

The behavior of Kconfig 'default' properties in Zephyr has changed. The 
earliest default with a satisfied condition is now used, instead of the 
last one. This is standard Kconfig behavior.

See http://docs.zephyrproject.org/porting/board_porting.html#old-zephyr-kconfig-behavior-for-defaults.

To get rid of this note, create a file called 'hide-defaults-note' in the 
Zephyr root directory. An empty file is fine.
-- Configuring done
-- Generating done
-- Build files have been written to: /root/zephyr/samples/hello_world/build
[1/85] Preparing syscall dependency handling

[2/85] Generating include/generated/syscall_macros.h
[3/85] Generating misc/generated/syscalls_subdirs.trigger
[4/85] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h
[5/85] Generating include/generated/driver-validation.h
[6/85] Generating misc/generated/syscalls.json
[7/85] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[8/85] Building C object zephyr/CMakeFiles/offsets.dir/arch/xtensa/core/offsets/offsets.c.obj
[9/85] Linking C static library zephyr/liboffsets.a
[10/85] Generating include/generated/offsets.h
[11/85] Building C object zephyr/CMakeFiles/zephyr.dir/arch/xtensa/core/cpu_idle.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/arch/xtensa/core/cpu_idle.c.obj 
/root/esp/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc -DBUILD_VERSION=v1.13.0-rc2-6-g493432393 -DKERNEL -D_FORTIFY_SOURCE=2 -D__ZEPHYR__=1 -I../../../kernel/include -I../../../arch/xtensa/include -I../../../arch/xtensa/soc/esp32 -I../../../arch/xtensa/soc/esp32/include -I../../../arch/xtensa/soc/include -I../../../boards/xtensa/esp32 -I../../../include -I../../../include/drivers -Izephyr/include/generated -I../../../lib/libc/minimal/include -I/root/esp/esp-idf/components/esp32/include -I/root/esp/esp-idf/components/soc/esp32/include -isystem /root/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/include -isystem /root/esp/xtensa-esp32-elf/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/include-fixed -Os -g -Wall -Wformat -Wformat-security -Wno-format-zero-length -imacros /root/zephyr/samples/hello_world/build/zephyr/include/generated/autoconf.h -ffreestanding -Wno-main -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -Wno-pointer-sign -Wno-unused-but-set-variable -fno-reorder-functions -fno-defer-pop -Werror=implicit-int -ffunction-sections -fdata-sections -mlongcalls -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/arch/xtensa/core/cpu_idle.c.obj -MF zephyr/CMakeFiles/zephyr.dir/arch/xtensa/core/cpu_idle.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/arch/xtensa/core/cpu_idle.c.obj   -c /root/zephyr/arch/xtensa/core/cpu_idle.c
/root/zephyr/arch/xtensa/core/cpu_idle.c:6:32: fatal error: xtensa/tie/xt_core.h: No such file or directory
compilation terminated.
[12/85] Building C object zephyr/CMakeFiles/zephyr.dir/arch/common/isr_tables.c.obj
[13/85] Building C object CMakeFiles/app.dir/src/main.c.obj
[14/85] Building C object zephyr/CMakeFiles/zephyr.dir/arch/xtensa/soc/esp32/soc.c.obj
[15/85] Building C object zephyr/CMakeFiles/zephyr.dir/arch/xtensa/soc/esp32/esp32-mp.c.obj
[16/85] Building C object zephyr/CMakeFiles/zephyr.dir/arch/xtensa/core/fatal.c.obj
ninja: build stopped: subcommand failed.
The command '/bin/sh -c cd $ZEPHYR_BASE/samples/hello_world &&     mkdir build && cd build &&     cmake -GNinja -DBOARD=esp32 .. &&     ninja flash' returned a non-zero code: 1
```

--- a/arch/xtensa/core/cpu_idle.c
+++ b/arch/xtensa/core/cpu_idle.c
@@ -3,8 +3,6 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <xtensa/tie/xt_core.h>
-#include <xtensa/tie/xt_interrupt.h>
 #include <tracing.h>
 
 /*

